pkg/commands/loaders/branches.go

```diff
@@ -114,77 +115,84 @@ func (self *BranchLoader) obtainBranches() []*models.Branch {
 
 	trimmedOutput := strings.TrimSpace(output)
 	outputLines := strings.Split(trimmedOutput, "\n")
-	branches := make([]*models.Branch, 0, len(outputLines))
-	for _, line := range outputLines {
+
+	branches := slices.FilterMap(outputLines, func(line string) (bool, *models.Branch) {
 		if line == "" {
-			continue
+			return false, nil
 		}
 
 		split := strings.Split(line, SEPARATION_CHAR)
 		if len(split) != 4 {
 			// Ignore line if it isn't separated into 4 parts
 			// This is probably a warning message, for more info see:
 			// https://github.com/jesseduffield/lazygit/issues/1385#issuecomment-885580439
-			continue
+			return false, nil
 		}
 
-		name := strings.TrimPrefix(split[1], "heads/")
-		branch := &models.Branch{
-			Name:      name,
-			Pullables: "?",
-			Pushables: "?",
-			Head:      split[0] == "*",
-		}
+		return true, parseBranch(split)
+	})
 
-		upstreamName := split[2]
-		if upstreamName == "" {
-			// if we're here then it means we do not have a local version of the remote.
-			// The branch might still be tracking a remote though, we just don't know
-			// how many commits ahead/behind it is
-			branches = append(branches, branch)
-			continue
-		}
+	return branches
+}
 
-		track := split[3]
-		re := regexp.MustCompile(`ahead (\d+)`)
-		match := re.FindStringSubmatch(track)
-		if len(match) > 1 {
-			branch.Pushables = match[1]
-		} else {
-			branch.Pushables = "0"
-		}
+func parseBranch(split []string) *models.Branch {
+	name := strings.TrimPrefix(split[1], "heads/")
+	branch := &models.Branch{
+		Name:      name,
+		Pullables: "?",
+		Pushables: "?",
+		Head:      split[0] == "*",
+	}
 
-		re = regexp.MustCompile(`behind (\d+)`)
-		match = re.FindStringSubmatch(track)
-		if len(match) > 1 {
-			branch.Pullables = match[1]
-		} else {
-			branch.Pullables = "0"
-		}
+	upstreamName := split[2]
+	if upstreamName == "" {
+		// if we're here then it means we do not have a local version of the remote.
+		// The branch might still be tracking a remote though, we just don't know
+		// how many commits ahead/behind it is
+		return branch
+	}
 
-		branches = append(branches, branch)
+	track := split[3]
+	re := regexp.MustCompile(`ahead (\d+)`)
+	match := re.FindStringSubmatch(track)
+	if len(match) > 1 {
+		branch.Pushables = match[1]
+	} else {
+		branch.Pushables = "0"
 	}
 
-	return branches
+	re = regexp.MustCompile(`behind (\d+)`)
+	match = re.FindStringSubmatch(track)
+	if len(match) > 1 {
+		branch.Pullables = match[1]
+	} else {
+		branch.Pullables = "0"
+	}
+
+	return branch
 }
 
 // TODO: only look at the new reflog commits, and otherwise store the recencies in
 // int form against the branch to recalculate the time ago
 func (self *BranchLoader) obtainReflogBranches(reflogCommits []*models.Commit) []*models.Branch {
-	foundBranchesMap := map[string]bool{}
+	foundBranches := set.New[string]()
```

Comment:
Having a dedicated set struct makes it much more explicit as to what you're doing (like with the above comparison with for loops, having an abstraction like this lets you know the intention up front rather than having to wait until you read some more code to understand what's going on)